applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp

```diff
@@ -379,6 +411,21 @@ void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::GetWakeDistance
     noalias(distances) = GetValue(WAKE_ELEMENTAL_DISTANCES);
 }
 
+template <int TDim, int TNumNodes>
+void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::GetEquationIdVectorExtendedElement(
+    EquationIdVectorType& rResult) const
+{
+    const int additional_node_id = GetAdditionalNode();
+
+    const auto& r_geometry = this->GetGeometry();
+    for (unsigned int i = 0; i < TNumNodes; i++)
+    {
+        rResult[i] = r_geometry[i].GetDof(VELOCITY_POTENTIAL).EquationId();
+    }
+    const auto& r_upwind_geometry = mpUpwindElement->GetGeometry();
+    rResult[TNumNodes + 1] = r_upwind_geometry[additional_node_id].GetDof(VELOCITY_POTENTIAL).EquationId();
```

Comment:
This is the most important suggestion, wanted to put it at the beginning but I forgot, related to what we already discussed
```suggestion
    rResult[TNumNodes] = r_upwind_geometry[additional_node_id].GetDof(VELOCITY_POTENTIAL).EquationId();
```